### PR TITLE
Clean up wording, bulleted lists, and code snippets in raspi-config and noninteractive section

### PR DESCRIPTION
**Pull request by nathan-contino**
* Brings syntax highlighting to these snippets, which were previously unmarked.
* Adds information about NVMe boot to the noninteractive boot order options list.
* Clarifies previously confusing noninteractive wifi connection instruction.

### CODE REVIEW

File: documentation/asciidoc/computers/configuration/raspi-config.adoc [hunk 1]
```diff
@@ -1,35 +1,32 @@
 [[raspi-config]]
-== The `raspi-config` Tool
+== `raspi-config`
 
-`raspi-config` is the Raspberry Pi configuration tool, originally written by https://github.com/asb[Alex Bradbury]. To open the configuration tool, type the following on the command line:
+TIP: Raspberry Pi Desktop users can also access a graphical version of this application at **Preferences** > **Raspberry Pi Configuration**.
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/raspi-config.adoc:4]: (although there are some "advanced options" which are only in the CLI tool and not the GUI tool)

File: documentation/asciidoc/computers/configuration/raspi-config.adoc [hunk 2]
```diff
@@ -170,12 +165,12 @@ Enable or disable a read-only filesystem.
 
 ===== Fan
 
-Set the behaviour of a GPIO connected fan.
+Customise the behaviour of a GPIO connected fan.
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/raspi-config.adoc:168]: Should "GPIO connected fan" be ""GPIO-connected fan" ? Might be worth clarifying that (AFAIK) this is only for use by https://www.raspberrypi.com/products/raspberry-pi-4-case-fan/

File: documentation/asciidoc/computers/configuration/raspi-config.adoc [hunk 3]
```diff
@@ -203,14 +198,12 @@ This option sets the country code for your wireless network.
 
 image::images/raspi-adv.png[raspi-config advanced options]
 
-NOTE: The options documented here will change depending on the model of Raspberry Pi that you're using, and whether you're using the Wayland or X11 backend. In the current version of Raspberry Pi OS Bookworm, the Raspberry Pi 4, Pi 400 and Pi 5 use Wayland by default; other models of Raspberry Pi use X11 by default.
-
 [[expand-filesystem]]
 ===== Expand filesystem
 
-This option will expand your installation to fill the whole SD card, giving you more space to use for files. You will need to reboot the Raspberry Pi to make this available. 
+Expands your installation to fill the entire storage device, giving you more space to use for files. You must reboot the Raspberry Pi to make this available.
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/raspi-config.adoc:204]: Raspberry Pi OS _should_ do this automatically, during its first-boot process. I think the option is only in `raspi-config` as a fail-safe fallback? (or if you e.g. use the "SD Card Copier" to clone your OS to a larger storage-medium)

File: documentation/asciidoc/computers/configuration/raspi-config.adoc [hunk 4]
```diff
@@ -247,364 +239,404 @@ Update this tool to the latest version.
...
 
-The `sudo` is required because you will be changing files that are not owned by the default user.
-
-NOTE: There is no consistent meaning for `0` and `1` in arguments. Each function will document what `0` and `1` mean for that particular function.
+NOTE: The meaning of `0` and `1` varies between options. Always check the documentation before passing a value to an option.
 
 [[raspi-config-cli-commands]]
 === List of options
 
-NOTE: The `raspi-config` tool is under continuous development, and the list of options below may not be completely up to date. Please be aware that different models of Raspberry Pi may have different options available.
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/raspi-config.adoc:278]: Might be worth retaining this warning? Or it least adding the "Please be aware that different models of Raspberry Pi may have different options available." to the previous NOTE?

File: documentation/asciidoc/computers/configuration/raspi-config.adoc [hunk 5]
```diff
@@ -247,364 +239,404 @@ Update this tool to the latest version.
...
 ==== System options
 
 ===== Wireless LAN
 
 Allows setting of the wireless LAN SSID and passphrase.
 
+[source,console]
 ----
-sudo raspi-config nonint do_wifi_ssid_passphrase <ssid> <passphrase> [hidden] [plain]
+$ sudo raspi-config nonint do_wifi_ssid_passphrase <ssid> <passphrase> [hidden] [plain]
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/raspi-config.adoc:272]: Perhaps we should remove the `raspi-config nonint do_wifi_ssid_passphrase` documentation, and just refer people to the `nmcli` docs instead? https://www.raspberrypi.com/documentation/computers/configuration.html#using-the-command-line :shrug:

File: documentation/asciidoc/computers/configuration/raspi-config.adoc [hunk 6]
```diff
@@ -247,364 +239,404 @@ Update this tool to the latest version.
...
 
-- `0` - bcm2835 Headphones
-- `1` - vc4-hdmi-0
-- `2` - vc4-hdmi-1
+* `0`: bcm2835 headphone jack
+* `1`: vc4-hdmi-0
+* `2`: vc4-hdmi-1
 
-NOTE: You may need to run the interactive version of `raspi-config` to determine the appropriate numbers to use with this option.
+For a full list of possible `<N>` values, see the abbreviations used in the interactive `raspi-config` version of this option.
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/raspi-config.adoc:318]: abbreviations -> numbers ?

File: documentation/asciidoc/computers/configuration/raspi-config.adoc (hunk 1 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/configuration/raspi-config.adoc:4]: I was trying to find the best way to bring this up. Probably an extra sentence here will do!

File: documentation/asciidoc/computers/configuration/raspi-config.adoc (hunk 2 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/configuration/raspi-config.adoc:168]: Ahhh excellent point, looks like the Active Cooler and Pi 5 case fan use a separate fan header. Thanks!

File: documentation/asciidoc/computers/configuration/raspi-config.adoc (hunk 3 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/configuration/raspi-config.adoc:204]: Worth mentioning for sure.

File: documentation/asciidoc/computers/configuration/raspi-config.adoc (hunk 4 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/configuration/raspi-config.adoc:278]: Continuous development is so common at this point that it's no longer a great excuse for outdated doc -- if we're missing something, I hope someone opens an issue and calls it out!

As for the model differences, that's worth calling out in both the int and non-int sections. But probably not in an admonition :)

File: documentation/asciidoc/computers/configuration/raspi-config.adoc (hunk 5 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/configuration/raspi-config.adoc:272]: I tested it today and it does work. Some of the errors are a bit wonky when you use nmcli, though -- I suspect some code paths might assume `wpa_supplicant` exists even when using nmcli.

I'm tempted to just point users to nmcli because of these issues, but as long as the option exists and works, some users might prefer it.

@spl237 , should we continue supporting this option? I'd be happy to open an issue on the `raspi-config` repo to either remove it or fix it. Your call.

File: documentation/asciidoc/computers/configuration/raspi-config.adoc (hunk 6 above)
Comment (nathan-contino) [reply documentation/asciidoc/computers/configuration/raspi-config.adoc:318]: Good call 😆 . Classic copy/paste mistake!

File: documentation/asciidoc/computers/configuration/raspi-config.adoc (hunk 5 above)
Comment (spl237) [reply documentation/asciidoc/computers/configuration/raspi-config.adoc:272]: It's not something I ever use in the GUI; I think it is there for the benefit of people who use raspi-config from the command-line. So @XECDesign's call, not mine!

File: documentation/asciidoc/computers/configuration/raspi-config.adoc [hunk 7]
```diff
@@ -25,231 +23,229 @@ Press the **Right** arrow key or press **Tab** to access the `<Select>` and `<Fi
...
-===== Overlay file system
+==== Overlay file system
 
 Enable or disable a read-only filesystem.
 
-===== Fan
+==== Fan
 
-Customise the behaviour of a GPIO connected fan.
+Customise the behaviour of the GPIO-connected https://www.raspberrypi.com/products/raspberry-pi-4-case-fan/[Raspberry Pi 4 Case Fan]. Does not effect the fans in the https://www.raspberrypi.com/products/raspberry-pi-5-case/[Raspberry Pi 5 Case for Raspberry Pi 5] or https://www.raspberrypi.com/products/active-cooler/[Raspberry Pi 5 Active Cooler], which connect using a special four-pin fan header.
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/raspi-config.adoc:165]: Another "effect" -> "affect" needed here.

File: documentation/asciidoc/computers/configuration/raspi-config.adoc [hunk 8]
```diff
@@ -25,231 +23,229 @@ Press the **Right** arrow key or press **Tab** to access the `<Select>` and `<Fi
...
+=== Advanced options
 
 image::images/raspi-adv.png[raspi-config advanced options]
 
 [[expand-filesystem]]
-===== Expand filesystem
+==== Expand filesystem
 
-Expands your installation to fill the entire storage device, giving you more space to use for files. You must reboot the Raspberry Pi to make this available.
+Expands your OS partition to fill the whole storage device, giving you more space to use for files. Reboot your Raspberry Pi to complete this action. Normally, Raspberry Pi OS runs this action on first boot. This option can be useful if you clone your OS to a separate storage device with more storage than the original.
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/raspi-config.adoc:201]: Could change the last "storage" to "capacity" ?

File: documentation/asciidoc/computers/configuration/raspi-config.adoc [hunk 9]
```diff
@@ -622,9 +618,9 @@ $ sudo raspi-config nonint do_overlayfs <0/1>
 * `0`: enable overlay filesystem
 * `1`: disable overlay filesystem
 
-===== Fan
+==== Fan
 
-Set the behaviour of a GPIO connected fan.
+Customise the behaviour of the GPIO-connected https://www.raspberrypi.com/products/raspberry-pi-4-case-fan/[Raspberry Pi 4 Case Fan]. Does not effect the fans in the https://www.raspberrypi.com/products/raspberry-pi-5-case/[Raspberry Pi 5 Case for Raspberry Pi 5] or https://www.raspberrypi.com/products/active-cooler/[Raspberry Pi 5 Active Cooler], which connect using a special four-pin fan header.
```
Comment (lurch) [inline documentation/asciidoc/computers/configuration/raspi-config.adoc:626]: effect -> affect

File: documentation/asciidoc/computers/configuration/raspi-config.adoc (hunk 5 above)
Comment (XECDesign) [reply documentation/asciidoc/computers/configuration/raspi-config.adoc:268]: What wonky error messages are you seeing and what's not working? If there's an issue I can reproduce, I'm happy to fix it.

However, I've always said that the `nonint` functionality is not intended for the user, so it doesn't make much sense documenting it. The idea is there is that it's used by other Raspberry Pi tools so that if we switch from dhcpcd to nm or make any other change, we only need to support it in one place. That way other tools won't need to be rebuilt and they won't diverge in behaviour. However, individual nonint commands can change or be removed without notice.

The short version - in my opinion, yes it absolutely makes sense for the user to use nmcli directly.
